hyperdome_client/threads.py

```
@@ -72,20 +72,232 @@ def run(self):
             return
 
 
-class WebThread(QtCore.QThread):
+class TaskSignals(QtCore.QObject):
+    success = QtCore.pyqtSignal(str)
+    error = QtCore.pyqtSignal(str)
+
+
+class SendMessageTask(QtCore.QRunnable):
     """
-    Starts the web service
+    send client message to server
     """
-    success = QtCore.pyqtSignal()
-    error = QtCore.pyqtSignal(str)
+    signals = TaskSignals()
 
-    def __init__(self, mode):
-        super(WebThread, self).__init__()
-        self.mode = mode
-        self.mode.common.log('WebThread', '__init__')
+    def __init__(self,
+                 server: Server,
+                 session: requests.Session,
+                 uid: str,
+                 message: str):
+        super(SendMessageTask, self).__init__()
+        self.server = server
+        self.session = session
+        self.uid = uid
+        self.message = message
+
+    def run(self):
+        try:
+            send_message(self.server, self.session, self.uid, self.message)
+        except requests.RequestException:
+            self.signals.error.emit("Couldn't send message")
+
+
+class GetUidTask(QtCore.QRunnable):
+    """
+    Get a UID from the server
+    """
+    signals = TaskSignals()
+
+    def __init__(self,
+                 server: Server,
+                 session: requests.Session):
+        super(GetUidTask, self).__init__()
+        self.server = server
+        self.session = session
 
     def run(self):
-        self.mode.common.log('WebThread', 'run')
-        self.mode.app.choose_port()
-        self.mode.web.start(self.mode.app.port,
-                            self.mode.app.stay_open)
+        try:
+            uid = get_uid(self.server, self.session)
+            self.signals.success.emit(uid)
+        except requests.RequestException:
+            self.signals.error.emit("couldn't get UID")
```

Same thing with swallowing exceptions

---

Yep. will create an issue for tracking.